cmd/query/app/server_test.go

```diff
@@ -602,12 +602,11 @@ func TestServerInUseHostPort(t *testing.T) {
 func TestServerSinglePort(t *testing.T) {
 	flagsSvc := flags.NewService(ports.QueryAdminHTTP)
 	flagsSvc.Logger = zaptest.NewLogger(t)
-	hostPort := ports.GetAddressFromCLIOptions(ports.QueryHTTP, "")
 	querySvc := makeQuerySvc()
 	server, err := NewServer(flagsSvc.Logger, flagsSvc.HC(), querySvc.qs, nil,
 		&QueryOptions{
-			GRPCHostPort: hostPort,
-			HTTPHostPort: hostPort,
+			GRPCHostPort: ":0",
+			HTTPHostPort: ":0",
```

Comment:
This will make the servers run on different random ports, would it not? It would invalidate the test.

Reply:
Thank you for pointing that out. You're right; changing the ports to "0" will cause the servers to run on different random ports, which could invalidate the test. I will revise the code to ensure that the servers run on the same port as intended.